Fix pulling in dependencies with composer

neclimdul opened the pull request:
Fixes #32

neclimdul committed:
Initial attempt at pulling in master deps with composer

msonnabaum commented on recipes/git.rb:
```diff
@@ -29,4 +29,24 @@
+      command "#{node['composer']['bin']} install --no-interaction --no-ansi --quiet --no-dev"
+      action :run
+      only_if "which composer"
+      not_if do
```
Lets just use the braces notation and fit this on one line.

msonnabaum commented on recipes/git.rb:
```diff
@@ -29,4 +29,24 @@
+      cwd node['drush']['install_dir']
+      command "#{node['composer']['bin']} install --no-interaction --no-ansi --quiet --no-dev"
+      action :run
+      only_if "which composer"
```
Why is this guard necessary? If we've included the recipe, are we not confident that composer is available?

msonnabaum replied:
Also, if this is necessary for some reason, i think I'd rather see a File.exists? on node['composer']['bin']

neclimdul committed:
Unneeded composer check

neclimdul committed:
Brace notation for not_if

neclimdul committed:
Don't run pear on master

neclimdul committed:
Merge remote-tracking branch 'upstream/master' into drush-composer

neclimdul committed:
Test git installation method

neclimdul committed:
Better git testing

neclimdul committed:
Merge remote-tracking branch 'origin/master' into drush-composer

neclimdul committed:
Fix test-kitchen

neclimdul committed:
Clean up ocmposer install

neclimdul committed:
Support alpha tags and beyond

neclimdul committed:
Add missing tests

neclimdul committed:
Update chef stuff

neclimdul committed:
Fix composer run

nicksantamaria commented:
This PR worked for me, nice work.

neclimdul committed:
Support setting drush to an individual commit

Also exclude some test suites that fail now because of the PHP version
requirements in drush.

neclimdul committed:
Refactor composer run

Try a different approach so recipes can use drush first run.
